analyzers/tests/SonarAnalyzer.Test/Rules/GenericLoggerInjectionShouldMatchEnclosingTypeTest.cs

```diff
@@ -0,0 +1,86 @@
+
+namespace SonarAnalyzer.Test.Rules;
+
+[TestClass]
+public class GenericLoggerInjectionShouldMatchEnclosingTypeTest
+{
+    private static readonly VerifierBuilder Builder = new VerifierBuilder<GenericLoggerInjectionShouldMatchEnclosingType>()
+        .AddReferences(NuGetMetadataReference.MicrosoftExtensionsLoggingAbstractions());
+
+    [TestMethod]
+    public void GenericLoggerInjectionShouldMatchEnclosingTypeTest_CS() =>
+        Builder.AddSnippet("""
+            using System;
+            using Microsoft.Extensions.Logging;
+
+            public class Correct : Base
+            {
+                Correct(ILogger<Correct> logger) { }                             // Compliant
+                Correct(ILogger<Correct> logger, ILogger<Correct> logger2) { }   // Compliant
+
+                Correct(Wrapper<Correct> logger) { }                             // Compliant
+                Correct(Wrapper<Wrong> logger) { }                               // Compliant
+                Correct(Wrapper<Correct> logger, Wrapper<Wrong> logger2) { }     // Compliant
+
+                Correct(ILogger<string> logger) { }                              // Noncompliant {{Update this logger to use its enclosing type.}}
+                //              ^^^^^^
+                Correct(ILogger<Wrong> logger) { }                               // Noncompliant
+                //              ^^^^^
+                Correct(ILogger<Base> logger) { }                                // Noncompliant
+                //              ^^^^
+
+                Correct(ILogger<Wrapper<Correct>> logger) { }                    // Noncompliant
+                //              ^^^^^^^^^^^^^^^^
+                Correct(ILogger<ILogger<Correct>> logger) { }                    // Noncompliant
+                //              ^^^^^^^^^^^^^^^^
+
+                Correct(ILogger<string> logger, ILogger<Wrong> logger2) { }
+                //              ^^^^^^ {{Update this logger to use its enclosing type.}}
+                //                                      ^^^^^ @-1 {{Update this logger to use its enclosing type.}}
+
+                Correct(ILogger<string> logger, ILogger<Correct> logger2, ILogger<ILogger<Correct>> logger3) { }
+                //              ^^^^^^ {{Update this logger to use its enclosing type.}}
+                //                                                                ^^^^^^^^^^^^^^^^ @-1 {{Update this logger to use its enclosing type.}}
+
+                Correct(Logger<Correct> logger, Logger<Wrong> logger2) { }
+                //                                     ^^^^^ {{Update this logger to use its enclosing type.}}
+
+                Correct(Logger logger) {} // Compliant FP , Logger is not a generic type
+            }
+
+            public class Base { }
+            public class Wrong { }
+            public class Wrapper<T> { }
+
+            public class Logger : Logger<int> { }
+
+            public class Logger<T> : ILogger<T>
```

Comment:
Could you please add a constructor to this class that receives a logger? Would be nice to have some test cases as well for generic classes.